src/EntityReferenceActionsHandler.php

```diff
@@ -184,18 +196,21 @@ public function formAlter(array &$element, FormStateInterface $form_state, array
   public function submitForm(array &$form, FormStateInterface $form_state) {
 
     $button = $form_state->getTriggeringElement();
-    $field_name = reset($button['#array_parents']);
 
-    /** @var \Drupal\Core\Entity\Display\EntityFormDisplayInterface $form_display */
-    $form_display = $form_state->getStorage()['form_display'];
+    $parents = array_slice($button['#array_parents'], 0, -2);
+    $field_name = end($parents);
+
+    $parents = array_slice($parents, 0, -1);
+    $sub_form = NestedArray::getValue($form, $parents);
```

Comment:
I think `$sub_form` should be called `$values`, because its values regardless of form structure / subform.